### PR TITLE
add linux default prefs, enable Jack, disable curl

### PR DESCRIPTION
**Pull request by KottV**
Hi,

This adds userprefs file for Linux, also enables Jack audio support and removes curl dependency

### CODE REVIEW

**Comment by KottV** [comment]
I wrote first attempt to address https://github.com/awwbees/BespokeSynth/issues/12

File: Source/UserData.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    #endif
+    juce::File localConfig (File::getSpecialLocation(File::userApplicationDataDirectory).getChildFile("BespokeSynth/data/userprefs.json").getFullPathName());
+    juce::File localBlank (File::getSpecialLocation(File::userApplicationDataDirectory).getChildFile("BespokeSynth/data/layouts/blank.json").getFullPathName());
+    
+    juce::File localConfigSave (File::getSpecialLocation(File::userApplicationDataDirectory).getChildFile("BespokeSynth/data/userprefs_save.json").getFullPathName());
+    juce::File localBlankSave (File::getSpecialLocation(File::userApplicationDataDirectory).getChildFile("BespokeSynth/data/layouts/blank_save.json").getFullPathName());
+    
+    if(juce::File(localDataDir).exists())
+    {
+        localConfig.copyFileTo(localConfigSave);
```
Comment (KottV) [inline Source/UserData.cpp:22]: I assume that `userprefs.json` and `layouts/blank.json` are user-modified files.

**Comment by KottV** [comment]
Nah, it doesn't work on JUCE 5.4.7

**Comment by awwbees** [comment]
I am taking a look at this. apologies for the slowness! I am pretty inexperienced with git/github and am still figuring out how to test and merge pull requests properly

**Comment by KottV** [comment]
I see!

You can make a `test` or `develop` branch where we can test commits.
Anyway this addition is completely safe :)

**Comment by KottV** [comment]
honestly this is my first JUCE code and of course it's better if you'll take a steady look on it

**Comment by awwbees** [comment]
ok! I've got a few changes I'm going to make, to better support having this behavior cross-platform like you had suggested in #12 

one question about a change in here: what was the purpose of disabling curl? does it cause some specific complication in linux?

**Comment by KottV** [comment]
>what was the purpose of disabling curl? does it cause some specific complication in linux?

dependencies in linux are headache
in Ubuntu it's libcurl-gnutls, in other distributives - libcurl, binary compiled in Ubuntu wont run on SUSE or Fedora without tricks
as curl is not really needed it is worth to just disable it

**Pull request merged**